crates/wasi-common/yanix/src/fcntl.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{
+    file::{FdFlag, OFlag},
+    Errno, Result,
+};
+use std::os::unix::prelude::*;
+
+pub fn dup_fd(fd: RawFd, close_on_exec: bool) -> Result<RawFd> {
```

Comment:
I am open to other approaches here, but we're currently converging on a convention that functions which operate on raw file descriptors should be marked unsafe, in part because that's what Rust's libstd does, and in part because that happens to align with WASI in a nice way -- it makes I/O unforgeable outside of unsafe code.

Reply:
Ahhh, I've overlooked that one. I agree 100% with the current convention on marking fns like this one `unsafe`. It's done now! :-)